src/applications/gi/components/profile/CalculatorForm.jsx

```diff
@@ -118,6 +119,72 @@ class CalculatorForm extends React.Component {
   renderYellowRibbon() {
     if (!this.props.displayedInputs.yellowRibbon) return null;
 
+    if (__BUILDTYPE__ !== 'production') {
+      let {
+        yellowRibbonDegreeLevelOptions,
+        yellowRibbonDivisionOptions,
+      } = this.props.inputs;
+
+      yellowRibbonDegreeLevelOptions =  yellowRibbonDegreeLevelOptions.map(value => ({ value, label: value }));
+      yellowRibbonDivisionOptions = yellowRibbonDivisionOptions.map(value => ({ value, label: value }));
+
+      return (
+        <div>
+          <RadioButtons
+            label={this.renderLearnMoreLabel({
+              text: 'Will you be a Yellow Ribbon recipient?',
+              modal: 'calcYr'
+            })}
+            name="yellowRibbonRecipient"
+            options={[
+              { value: 'yes', label: 'Yes' },
+              { value: 'no', label: 'No' }
+            ]}
+            value={this.props.inputs.yellowRibbonRecipient}
+            onChange={this.handleInputChange}/>
+          { this.props.inputs.yellowRibbonRecipient === 'yes' ?
+            <div>
+              <Dropdown
+                label="Degree Level"
+                name={'yellowRibbonDegreeLevel'}
+                alt="Degree Level"
+                options={yellowRibbonDegreeLevelOptions}
+                visible
+                value={this.props.inputs.yellowRibbonDegreeLevel}
+                onChange={this.handleInputChange}/>
+              <Dropdown
+                label="Division or school"
+                name={'yellowRibbonDivision'}
+                alt="Division or school"
+                options={yellowRibbonDivisionOptions}
+                visible
+                value={this.props.inputs.yellowRibbonDivision}
+                onChange={this.handleInputChange}/>
+              <div>
+                <label htmlFor="yellowRibbonAmount">
+                  Yellow Ribbon amount from school per year
+                </label>
+                <input
+                  type="text"
+                  name="yellowRibbonAmount"
+                  value={formatCurrency(this.props.inputs.yellowRibbonAmount)}
+                  onChange={this.handleInputChange}/>
+              </div>
+              <AlertBox
+                isVisible
+                key={this.props.inputs.yellowRibbonProgramIndex}
```

Comment:
Doesn't update when the children change. Decided to add an index to the yellowPrograms fetched and to use that as a key.